partner_ref_unique/models/res_partner.py

```diff
@@ -22,7 +22,7 @@ def _check_ref(self):
                 ]
                 if mode == "companies":
                     domain.append(("is_company", "=", True))
-                other = self.search(domain)
+                other = self.with_context(active_test=False).search(domain)
```

Comment:
I think the original behaviour is intended in order to archive partner without deleting it (to keep history).

Your test shows that you should unlink() the partner in order to create a new one with the same ref

Reply:
@rousseldenis original behavior allows to have 2 partners w/ the same ref as soon as one of them is archived. With this code, we test that all the unicity of the ref is among all partners (archived or not).

About the test, we mimic the same 'unlink' than on the original tests. 
In the `test_partner1_archived_new_partner_same_ref` test, we archive the partner1 (line 70) and then try to create a new one w/ the same ref. 

Maybe/probably I misunderstood your comment :smile:

Reply:
> @rousseldenis original behavior allows to have 2 partners w/ the same ref as soon as one of them is archived. With this code, we test that all the unicity of the ref is among all partners (archived or not).

Yes I understood the purpose of this.

My question is about the original behaviour. Maybe was it at purpose to have archived partners with same reference than the active one.

Your change is not a fix but a module behavior change